devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java

```diff
@@ -89,7 +89,8 @@ public void prepareQuarkus() {
                         paths.iterator().next(),
                         buildDir,
                         sourceRegistrar,
-                        appCreationContext.getAppModel());
+                        appCreationContext.getAppModel(),
+                        getBuildSystemProperties(extension().getAppArtifact()));
```

Comment:
Looks like we could re-use `realProperties` instead of re-building them again?

Reply:
+1

Reply:
done